WearTilesKotlin/app/src/main/java/com/example/wear/tiles/messaging/MessagingTileRenderer.kt

```diff
@@ -35,69 +36,56 @@ import androidx.wear.tiles.LayoutElementBuilders.VERTICAL_ALIGN_CENTER
 import com.example.wear.tiles.R
-import com.example.wear.tiles.singleEntryTimeline
-import com.example.wear.tiles.util.TileRenderer
+import com.google.android.horologist.tiles.SingleTileLayoutRenderer
 
 class MessagingTileRenderer(context: Context) :
-    TileRenderer<MessagingTileState, Map<Contact, Bitmap>>(context) {
+    SingleTileLayoutRenderer<MessagingTileState, Map<Contact, Bitmap>>(context) {
+
     override fun renderTile(
-        tileState: MessagingTileState,
-        requestParams: RequestBuilders.TileRequest,
-    ): Tile {
-        return Tile.Builder()
-            .setResourcesVersion(PERMANENT_RESOURCES_VERSION)
-            // Creates a timeline to hold one or more tile entries for a specific time periods.
-            .setTimeline(
-                singleEntryTimeline(
-                    tileLayout(tileState, requestParams.deviceParameters!!)
-                )
-            ).build()
+        singleTileState: MessagingTileState,
+        deviceParameters: DeviceParametersBuilders.DeviceParameters
+    ): LayoutElement {
+        return tileLayout(singleTileState, deviceParameters)
     }
 
-    override fun produceRequestedResources(
+    override fun Resources.Builder.produceRequestedResources(
         resourceResults: Map<Contact, Bitmap>,
-        requestParams: RequestBuilders.ResourcesRequest
-    ): Resources {
-        val resourcesBuilder = Resources.Builder().setVersion(PERMANENT_RESOURCES_VERSION)
-
-        if (requestParams.resourceRequested(ID_IC_SEARCH)) {
-            resourcesBuilder.addIdToImageMapping(
+        deviceParameters: DeviceParametersBuilders.DeviceParameters,
+        resourceIds: MutableList<String>
+    ) {
+        if (resourceIds.isEmpty() || resourceIds.contains(ID_IC_SEARCH)) {
+            addIdToImageMapping(
                 ID_IC_SEARCH,
-                ImageResource.Builder()
-                    .setAndroidResourceByResId(
-                        ResourceBuilders.AndroidImageResourceByResId.Builder()
-                            .setResourceId(R.drawable.ic_search)
-                            .build()
-                    )
-                    .build()
+                imageResourceFrom(R.drawable.ic_search)
             )
         }
 
         // Add the scaled & cropped avatar images
         resourceResults.forEach { (contact, bitmap) ->
             val imageResource = bitmapToImageResource(bitmap)
             // Add each created image resource to the list
-            resourcesBuilder.addIdToImageMapping(
+            addIdToImageMapping(
                 "$ID_CONTACT_PREFIX${contact.id}",
                 imageResource
             )
         }
-
-        return resourcesBuilder.build()
     }
 
-    private fun RequestBuilders.ResourcesRequest.resourceRequested(resourceId: String) =
-        resourceIds.isEmpty() || resourceIds.contains(resourceId)
+    private fun imageResourceFrom(@DrawableRes resourceId: Int) = ImageResource.Builder()
```

Comment:
Is this worth adding to Horologist? Either standalone function or an extension on Resources.Builder?

Reply:
Yes, I added an issue for it :D https://github.com/google/horologist/issues/229